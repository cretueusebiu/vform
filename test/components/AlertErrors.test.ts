import { mount } from '@vue/test-utils'
import { Form, AlertErrors } from 'vform'

describe('AlertErrors', () => {
  test('render alert with a list of errors', () => {
    const form = new Form({ username: '', password: '' })
    form.errors.set({
      username: ['Username is required'],
      password: ['Password must be at least 6 characters', 'Password does not match']
    })

    const wrapper = mount(AlertErrors, {
      propsData: { form, message: 'Custom message' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('dismiss alert', () => {
    const form = new Form({ username: '', password: '' })
    form.errors.set({
      username: ['Username is required'],
      password: ['Password must be at least 6 characters', 'Password does not match']
    })

    const wrapper = mount(AlertErrors, {
      propsData: { form, message: 'Custom message' }
    })

    wrapper.find('.close').trigger('click')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
