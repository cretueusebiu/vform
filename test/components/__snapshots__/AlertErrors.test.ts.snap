// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertErrors dismiss alert 1`] = `
<div role="alert" class="alert alert-danger alert-dismissible"><button type="button" aria-label="Close" class="close"><span aria-hidden="true">×</span></button>
  <div>Custom message</div>
  <ul>
    <li>Username is required</li>
    <li>Password must be at least 6 characters</li>
    <li>Password does not match</li>
  </ul>
</div>
`;

exports[`AlertErrors render alert with a list of errors 1`] = `
<div role="alert" class="alert alert-danger alert-dismissible"><button type="button" aria-label="Close" class="close"><span aria-hidden="true">×</span></button>
  <div>Custom message</div>
  <ul>
    <li>Username is required</li>
    <li>Password must be at least 6 characters</li>
    <li>Password does not match</li>
  </ul>
</div>
`;
